scripts/util.py

```diff
@@ -153,6 +155,66 @@
 ] = lambda self, data: yaml.representer.SafeRepresenter.represent_str(self, str(data))
 
 
+class ApiEndpoint(Enum):
+    COMPUTE = "compute"
+    BQ = "bq"
+    STORAGE = "storage"
+    TPU = "tpu"
+    SECRET = "secret_manager"
+
+
+_ud = "googleapis.com"
+
+
+def universe_domain(refresh: bool = False) -> str:
+    global _ud
+    if refresh:
+        _ud = instance_metadata("attributes/universe_domain")
+    return _ud
+
+
+def endpoint_version(api: ApiEndpoint):
+    if api.value in lkp.endpoint_versions:
+        return lkp.endpoint_versions[api.value]
+    return None
+
+
+def get_credentials(credentials=None) -> service_account.Credentials:
+    try:
+        key_path = os.environ["GOOGLE_APPLICATION_CREDENTIALS"]
+    except KeyError:
+        key_path = None
+    if key_path is not None:
+        credentials = service_account.Credentials.from_service_account_file(
+            key_path, scopes=[f"https://www.{universe_domain()}/auth/cloud-platform"]
+        )
+    elif credentials is None:
+        credentials = def_creds
```

Comment:
I would advice to remove global `def_creds`

Reply:
Moved to function.